### DISCUSSION TITLE
AIR apps show Let's Encrypt certificate expired also on newer Androids

### DISCUSSION THREAD

**Discussion by pfoof** [airsdk/Adobe-Runtime-Support#1234] [Q&A]
So as you know on Sept 30 X3 certificate expired and X1 was not added until Nougat 7.1. However, our apps started popping up certificate warnings even on later Android versions like Oreo or 11.

Does AIR use system-bundled certificate store or it's own? The message definitely comes from AIRSDK as I found it in `strings.xml`.

**Comment by ajwfrost** [comment]
AIR should use the system certificate store (in /system/etc/security/)

Are you able to provide a build of your app with "apk-debug" so that we can test it here - or just a URL that we can test against (are you using http/loader requests, or secure sockets, ...?)

thanks

**Reply by pfoof** [reply]
It's with `flash.net.URLLoader`, example `https://avatars.spiele-palast.de`

**Comment by Michoko92** [comment]
Happened to me too. Since my host didn't want to do anything about this, I'm in the process of republishing all my apps, compiled with a more recent version of AIR (which includes an updated OpenSSL layer). Such a pain, but didn't find another solution...

**Comment by pfoof** [comment]
Yes, it's rather on AIR side, I could reproduce the issue on KitKat in our app and not in the stock browser or Chrome

**Comment by ajwfrost** [comment]
So just to make sure we're all seeing the same thing on Android:

1) an 'old' version of AIR, pre-33.1.1.476, will display the certificate error dialog when it encounters this certificate -> users can decide to trust the certificate for this session although there's perhaps no reason why they should!
2) when built with a newer version of AIR, the certificate is accepted
3) this is basically independent of the Android version i.e. it seems to be purely related to the version of the OpenSSL library within AIR

@pfoof are you able to confirm this i.e. when you're reproducing this on KitKat and on Oreo etc, you're using an 'old' version of AIR?

If anyone has an issue with the latest version of AIR then it's something we need to look into; if it's only happening with an older version then I'm not sure we'd be able to help, I don't know of any workarounds for this (that don't involve changing the certificate on the server..)

thanks

**Reply by pfoof** [reply]
So I just looked into our build artifacts and it is exactly 476. We will update it and get back with the result

**Reply by ajwfrost** [reply]
476 should include all the fixes here... we can double-check what we get with that build as well though, currently we just tried AIR 32 and our latest SDK..

**Reply by ajwfrost** [reply]
FYI that URL works for me using 33.1.1.476 without showing any certificate dialog... (when I say "works", I'm getting a 403 forbidden status response which is what I also get via browsers on mobile/desktop; with 32 I was getting the certificate dialog before this and then getting the 403 response if I accepted the certificate).
